generators/app/index.js

```diff
@@ -141,34 +213,73 @@ module.exports = generator.extend({
             // To access props later use this.props.someOption;
             done();
         });
-    },
+    }
 
     // write the generator-specific files
     writing() {
-        // replace files with Spring's naming strategies
-        this.log(chalk.bold.yellow('JHipster-db-helper replaces your naming strategies :'));
-        this._replaceNamingStrategies(jhipsterVar.jhipsterConfig.buildTool);
-
-        // declarations done by jhipster-module
-        this.baseName = jhipsterVar.baseName;
-        this.packageName = jhipsterVar.packageName;
-        this.angularAppName = jhipsterVar.angularAppName;
-        this.clientFramework = jhipsterVar.clientFramework;
-        this.clientPackageManager = jhipsterVar.clientPackageManager;
-        this.message = this.props.message;
-
-        try {
-            jhipsterFunc.registerModule('generator-jhipster-db-helper', 'app', 'post', 'app', 'A JHipster module for already existing databases');
-        } catch (err) {
-            this.log(`${chalk.red.bold('WARN!')} Could not register as a jhipster entity post creation hook...\n`);
-        }
+        const configFile = this._getConfigFilePath(this.dbhTestCase);
 
-        try {
-            jhipsterFunc.registerModule('generator-jhipster-db-helper', 'entity', 'post', 'fix-entity', 'A JHipster module to circumvent JHipster limitations about names');
-        } catch (err) {
-            this.log(`${chalk.red.bold('WARN!')} Could not register as a jhipster entity post creation hook...\n`);
-        }
-    },
+        this._getPolyfill(configFile)
```

Comment:
No, but in order to avoid race conditions, all logic that relies on the fulfilling of a Promise must become thenable (go inside a Promise.then method). When Node v7 becomes LTS, we could refactor this with `async`/`await` and have a better looking code.